### PR TITLE
CairoRunner.run_until_pc_with_steps_limit

### PR DESCRIPTION
**Pull request by pefontana**
# Add CairoRunner.run_until_pc_with_steps_limit method

Description of the pull request changes and motivation.

## Checklist
- [ ] Linked to Github Issue
- [x] Unit tests added
- [x] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [x] CHANGELOG has been updated.

### CODE REVIEW

**Review by fmoletta — COMMENTED** [review]
Aren't `run_until_steps` and `run_for_steps` already satisfying this functionality?

**Comment by pefontana** [comment]
> Aren't `run_until_steps` and `run_for_steps` already satisfying this functionality?

They are similar but no

run_for_steps returns an Error if you finish executing the program and OK if the execution didn't finish
run_until_pc_with_steps_limit returns OK if you finish executing the Program, and Error if you don't finish the execution

**Review by fmoletta — APPROVED** [review]

File: src/vm/errors/vm_errors.rs [hunk 1]
```diff
@@ -93,6 +93,8 @@ pub enum VirtualMachineError {
     NoImm,
     #[error("Execution reached the end of the program. Requested remaining steps: {0}.")]
     EndOfProgram(usize),
+    #[error("Couldnt reached the end of the program. Executed steps: {0}.")]
+    StepsLimit(u64),
```
Comment (juanbono) [inline src/vm/errors/vm_errors.rs:97]: are we sure that u64 is enough?

**Review by juanbono — APPROVED** [review]

**Review by juanbono — APPROVED** [review]

File: src/vm/errors/vm_errors.rs (hunk 1 above)
Comment (igaray) [reply src/vm/errors/vm_errors.rs:97]: Couldnt -> Could not (formal english please)

Comment (pefontana) [reply src/vm/errors/vm_errors.rs:97]: I think u64 will be enough, u64::MAX = 18_446_744_073_709_551_615u64
Also `current_step ` and the other methods uses `usize` for the step counters

**Pull request merged**